### PR TITLE
[VITE] Missing types and type issues

### PR DESCRIPTION
**Pull request by rubenthoms**

### CODE REVIEW

**Comment by hkfb** [comment]
@t0oF-azpn for me our cypress tests are hanging when this change is applied. Can you reproduce it?

File: react/src/lib/components/SubsurfaceViewer/layers/wells/wellsLayer.ts [hunk 1]
```diff
@@ -669,7 +669,8 @@ WellsLayer.defaultProps = {
 //================= Local help functions. ==================
 
 function getColumn<D>(data: D[][], col: number): D[] {
-    const column = [];
+    // @rmt: Missing type
```
Comment (t0oF-azpn) [inline react/src/lib/components/SubsurfaceViewer/layers/wells/wellsLayer.ts:672]: Remove comment

**Comment by t0oF-azpn** [comment]
> @t0oF-azpn for me our cypress tests are hanging when this change is applied. Can you reproduce it?

No problem on my side.

BTW, many cypress tests are not compiling. I could go back to Feb 17th, they were already failing:
https://github.com/equinor/webviz-subsurface-components/actions/runs/4206070011/jobs/7299046950?pr=1426
(note, the logs are only kept for 3 months)

File: react/src/lib/components/ColorLegends/WebVizColorLegend.tsx [hunk 2]
```diff
@@ -4,7 +4,8 @@ import { ColorLegend } from "@emerson-eps/color-tables";
 import { colorTablesArray } from "@emerson-eps/color-tables/";
 
 interface LegendProps {
-    colorTables?: colorTablesArray | string;
+    // @rmt: Not matching the type in @emerson-eps/color-tables
```
Comment (t0oF-azpn) [inline react/src/lib/components/ColorLegends/WebVizColorLegend.tsx:7]: Fine for me, you can remove the comments

File: react/src/lib/components/SubsurfaceViewer/components/DistanceScale.tsx [hunk 3]
```diff
@@ -28,7 +28,10 @@ const DistanceScale: React.FC<ScaleProps> = ({
     style,
     scaleUnit,
 }: ScaleProps) => {
-    if (!zoom || !widthPerUnit || !incrementValue) return null;
+    // @rmt: added scaleUnit check - NOTE: if any of the values below === 0 || === "", this will return null
+    if (!zoom || !widthPerUnit || !incrementValue || scaleUnit === undefined) {
```
Comment (t0oF-azpn) [inline react/src/lib/components/SubsurfaceViewer/components/DistanceScale.tsx:32]: Can't you test `convert(0).getUnit(scaleUnit)` to make sure the unit does indeed exist ?

If scaleUnits is not valid, the conversion will throw :/

File: react/src/lib/components/SubsurfaceViewer/SubsurfaceViewer.tsx [hunk 4]
```diff
@@ -115,8 +115,9 @@ const SubsurfaceViewer: React.FC<SubsurfaceViewerProps> = ({
             return;
         }
 
-        const enumerations = [];
-        const layersJson = layers as unknown;
+        // @rmt: Added broad type - should be improved?
+        const enumerations: Record<string, unknown>[] = [];
```
Comment (t0oF-azpn) [inline react/src/lib/components/SubsurfaceViewer/SubsurfaceViewer.tsx:119]: I'm fine with this.

**Comment by hkfb** [comment]
> > @t0oF-azpn for me our cypress tests are hanging when this change is applied. Can you reproduce it?
> 
> No problem on my side.
> 
> BTW, many cypress tests are not compiling. I could go back to Feb 17th, they were already failing: https://github.com/equinor/webviz-subsurface-components/actions/runs/4206070011/jobs/7299046950?pr=1426 (note, the logs are only kept for 3 months)

@t0oF-azpn I meant the internal cypress tests in the `viewers-clients` repo.

**Comment by t0oF-azpn** [comment]
> > > @t0oF-azpn for me our cypress tests are hanging when this change is applied. Can you reproduce it?
> > 
> > 
> > No problem on my side.
> > ...
> 
> @t0oF-azpn I meant the internal cypress tests in the `viewers-clients` repo.

Same problem in our repo. There is one signle test stalling. If I skip this test, the run is fine.
I'm investigating it...

**Comment by t0oF-azpn** [comment]
The only test stalling is using `getCameraPosition` callback.
If I remove this callback, the test runs fine.

The test stalls because of re-entrant function:
image

**Comment by anders-kiaer** [comment]
@t0oF-azpn @hkfb Are you okey with we get this in and do a alpha-release, and you later add a new test upstream here reproducing the problem you see internally? Easier to collaborate on an open test failing. 🙂

**Comment by t0oF-azpn** [comment]
I'm OK with that. Otherwise, it'll take ages to get it through :/

**Comment by t0oF-azpn** [comment]
BTW, havard is off for 2 weeks (starting last week)...

**Pull request merged**